test-basic/data-movement-request.js

```diff
@@ -20,26 +20,21 @@ const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection)
 
 const Stream = require('stream');
 const readable = new Stream.Readable({objectMode: true});
-const writable = new Stream.Writable({objectMode: true});
+const batchSize = 10;
 
 describe('data-movement-requests test', function(){
 
     it('should write 1000 documents', function(done){
 
-        writable._write = (object, encoding, done) => {
-            done();
-        };
-
-        readable.pipe(writable);
-        createDocumentDescriptors();
-        readable.on('readable', function() {
-            writeDocs(readable.read(), done);
-        });
+        createInput();
+        for(var i=0; i<3; i++) {
+            writeDocs(i, done);
+        }
         done();
```

Comment:
Calling done() after starting the requests means the test will finish before the stream is read.

Instead, done() should be called only after the last batch finishes writing.

Reply:
Hi Erik, I did not remove this done() since moving this makes the test throw the error - 
Error: done() called multiple times in test <data-movement-requests test should write 1000 documents> of file

Reply:
> Error: done() called multiple times in test

That indicates that the finalization after the last response isn't working correctly because it is calling done() more than once. 

Calling done() after the loop on line 38 finishes the test before the first request has been sent to the server.

In a prototype, such problems don't matter much but

*  In the real implementation, it will be important to finalize only after the last response.
*  In the real test, it will be necessary to call done() only after the last response finishes so that the final result can be inspected.
 
Whether to work out the fix now in the prototype or in the real implementation is up to the implementer...